frontend/src/RecordingPageComp/RecordingSelection/RecordingSelectionComp.jsx

```diff
@@ -69,25 +76,36 @@ function RecordingSelector({ setSelectedBlob, selectedBlob, setIsLoadingRecordin
       } else {
         const data = await response.json();
         const blobs = data.map((item) => {
-          const audioBlob = convertToBlob(item.audios.data, "audio/mp3"); //converts each value in data.audios.data into a blob
+          const audioBlob = convertToBlob(item.audios.data, "audio/mp3");
           return audioBlob;
         });
-        setRecordings(blobs); //Updates state variable with recent list of recordings
+        setRecordings(blobs);
       }
     } catch (error) {
-      alert("Failed to fetch audios. Please try again.");
+      alert("Failed to fetch audios. Please try again.", error.message);
     }
   };
 
+  useEffect(() => {
+    const newBlobs = recordings.map((item) => {
+      return URL.createObjectURL(item);
+    });
+    setRecordingUrls(newBlobs);
+
+    return () => {
+      newBlobs.forEach((blobUrl) => URL.revokeObjectURL(blobUrl));
+    };
+  }, [recordings]);
+
```

Comment:
What does this do?

Reply:
As discussed in 1:1, removes URL from each blob to improve performance.